packages/core/src/agent/index.ts

```diff
@@ -220,8 +220,7 @@ export class Agent<
         return { threadId: threadId || '', messages: userMessages };
       }
 
-      const userMessage = this.getMostRecentUserMessage(userMessages);
-      const newMessages = userMessage ? [userMessage] : userMessages;
```

Comment:
this was also a problem that we were only using the latest user message since the latest message could be a tool call for client side tools